quickwit-indexing/src/controllable_directory.rs

```diff
@@ -0,0 +1,264 @@
+// Copyright (C) 2021 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::io::{BufWriter, IntoInnerError};
+use std::path::Path;
+use std::sync::Arc;
+use std::{fmt, io};
+
+use quickwit_actors::{KillSwitch, Progress};
+use tantivy::common::{AntiCallToken, TerminatingWrite};
+use tantivy::directory::error::{DeleteError, OpenReadError, OpenWriteError};
+use tantivy::directory::{FileHandle, WatchCallback, WatchHandle, WritePtr};
+use tantivy::Directory;
+
+/// Buffer capacity.
+///
+/// This is the current default for the BufWriter, but considering this constant
+/// will have a direct impact on health check, we'd better fix it.
+const BUFFER_NUM_BYTES: usize = 8_192;
```

Comment:
`8_192` sounds like a good starting value, do you want to update the buffer size regarding the number of bytes to write?

Reply:
No. 8KB or 16KB is considered optimal. 8K is the default in rust btw